### ISSUE TITLE
The C++20 break u8 string literal compatibility bug

### ISSUE THREAD

**Issue by uikino** [cycfi/elements#299]
# Bug 
The C++20 [P1423R2](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2019/p1423r2.html) paper  make compiler compile error if  source code using u8"..." string literal with ```-std=c++20```
# Suggestion
remove all u8 string literal prefix

**Comment by djowel** [comment]
Yes, this is a known issue. I'm not happy with suggestions yet that will satisfy both c++17 and c++20.

**Comment by djowel** [comment]
I am OK with this solution: http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2019/p1423r2.html#emulate
Are you willing to prepare a PR?

**Comment by uikino** [comment]
I work with Linux and do not have MSVC installed, so the solution provided in the PR can only be described as initially verified on GCC for Linux.

**Comment by zhou881117** [comment]
this is the reason upgrade to c++20？
maybe c++17  more suitable
